### ISSUE TITLE
ck progress seems stuck ,table cann't merge and connected ,then a lot of close_wait

### ISSUE THREAD

**Issue by weiyongyuan** [ClickHouse/ClickHouse#11326]
**Describe the bug**
I have a cluster consited of one distribute node and 2 shards 4 replicas.
I insert batch data into one os the replicas which is not leader at a speed about 80~130 million rows/second。abount 15 minutes later,the produce data progress can't insert data successfully,the error log shows too many parts because this replica doesn't merge data.At the same time ,this shard's leader replica merge data successfully.This replica'progress is alive but it seems struck. And  the clickhouse_client can't login in and the distribute query to the replica will be timeout and this replica has too many close_wait connections,
**How to reproduce**
It's hard to reproduce.I only meet several times in the past half year.You can test to insert into one of the replica at a very high speed.
* Which ClickHouse server version to use
20.1.6.30

**Expected behavior**
The  insert and merge of replica should work normally!

**Error message and/or stacktrace**
These are the screenshots i capture then. 
1.the picture shows the too many parts error:
![image](https://user-images.githubusercontent.com/20318636/83368560-d6867700-a3eb-11ea-89dd-c5cf6472d7ae.png)
2.this picture shows the replicas's every partition's active parts count:
![image](https://user-images.githubusercontent.com/20318636/83368675-4137b280-a3ec-11ea-929d-46d32e855695.png)
3.this picture shows the connections status:
![image](https://user-images.githubusercontent.com/20318636/83369105-c079b600-a3ed-11ea-969d-73740796b14d.png)

**Comment by weiyongyuan** [comment]
Yestoday i met this problem again,the below log shows that the inserts and merges all don't work now!
-rw-r----- 1 root root    1 Jun  2 14:12 format_version.txt
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79020_79020_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79021_79021_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79022_79022_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79023_79023_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79024_79024_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79025_79025_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79026_79026_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79027_79027_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79028_79028_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_insert_441968_79029_79029_0
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441966_0_2412_4
drwxr-x--- 2 root root 4096 Jun  2 16:10 tmp_merge_441966_13807_14060_3
drwxr-x--- 2 root root 4096 Jun  2 16:10 tmp_merge_441966_8504_8764_3
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441967_33439_33544_3
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441967_33714_33764_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_100_125_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_126_150_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_1729_1753_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_1805_1829_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_181_205_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_1830_1854_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_206_230_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_4039_4063_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_4069_4094_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_4441_4464_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_4465_4489_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_4566_4590_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_4591_4615_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_531_555_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_556_580_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_6854_6878_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_6924_6947_2
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7083_7091_1
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7092_7098_1
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7099_7108_1
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7132_7141_1
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7151_7159_1
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7170_7177_1
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7178_7184_1
drwxr-x--- 2 root root 4096 Jun  2 16:13 tmp_merge_441968_7185_7191_1

**Comment by weiyongyuan** [comment]
Today,i found that in the /proc/ckpid/fd,there ara so many socket created but not use.


lrwx------ 1 root root 64 Jun  6 10:01 994 -> socket:[80756151]
lrwx------ 1 root root 64 Jun  6 10:01 995 -> socket:[80756154]
lrwx------ 1 root root 64 Jun  6 10:01 996 -> socket:[80747433]
lrwx------ 1 root root 64 Jun  6 10:01 997 -> socket:[80747434]
lrwx------ 1 root root 64 Jun  6 10:01 998 -> socket:[80747435]
lrwx------ 1 root root 64 Jun  6 10:01 999 -> socket:[80747436]
[root@bd-server-8 ~]# ll /proc/73589/fd/|wc -l
4270
It seems a bug of CH.

**Comment by weiyongyuan** [comment]
Afterwards,we had debug clickhouse with gdb when it happened this issue.The stacktrace shows that clickhouse has a deadlock when i turn on the system.text_log.at the 98th line of  OwnSplitChannel.cpp 
```
 std::lock_guard<std::mutex> lock(text_log_mutex);
    if (auto log = text_log.lock())
        log->add(elem);
```
Here is the dump：
[a.zip](https://github.com/ClickHouse/ClickHouse/files/4958719/a.zip)
Now I turn off the system.text_log.
BTW，i've seen some bug fix about log thread race [，](https://github.com/ClickHouse/ClickHouse/pull/11314),i don't know if thes bug is related with this issue.

**Comment by MrSsunlight** [comment]
I also encountered the same problem, how did you solve your problem

**Comment by alexey-milovidov** [comment]
Upgrade to version 20.12 or newer.

**Comment by dahuilang3280** [comment]
> Afterwards,we had debug clickhouse with gdb when it happened this issue.The stacktrace shows that clickhouse has a deadlock when i turn on the system.text_log.at the 98th line of OwnSplitChannel.cpp
> 
> ```
>  std::lock_guard<std::mutex> lock(text_log_mutex);
>     if (auto log = text_log.lock())
>         log->add(elem);
> ```
> 
> Here is the dump： [a.zip](https://github.com/ClickHouse/ClickHouse/files/4958719/a.zip) Now I turn off the system.text_log. BTW，i've seen some bug fix about log thread race,(#11314 don't know if thes bug is related with this issue.

我也遇到类似的问题，想请教一下要如何定位到具体死锁的地方，能否给个联系方式私下请教一下
